examples/solids/qfunctions/hyperSS.h

```diff
@@ -0,0 +1,295 @@
+    for (int j = 0; j < 3; j++)     // Component
+      for (int k = 0; k < 3; k++) { // Derivative
+        graddeltau[j][k] = 0;
+        for (int m = 0; m < 3; m++)
+          graddeltau[j][k] += dXdx[m][k] * deltadu[j][m];
+      }
+
+    // Compute Strain : e (epsilon)
+    // e = 1/2 (grad u + (grad u)^T)
+    const CeedScalar de00 = (graddeltau[0][0] + graddeltau[0][0])/2.,
+                     de01 = (graddeltau[0][1] + graddeltau[1][0])/2.,
+                     de02 = (graddeltau[0][2] + graddeltau[2][0])/2.,
+                     de11 = (graddeltau[1][1] + graddeltau[1][1])/2.,
+                     de12 = (graddeltau[1][2] + graddeltau[2][1])/2.,
+                     de22 = (graddeltau[2][2] + graddeltau[2][2])/2.;
+    // *INDENT-OFF*
+    const CeedScalar de[3][3] =  {{de00, de01, de02},
+                                  {de01, de11, de12},
+                                  {de02, de12, de22}
+                                 };
+    // *INDENT-ON*
+    
+    //strain (epsilon)
+    //    and
+    //stress (sigma) in Voigt notation:
+    //            [e00]               [sigma00]
+    //            [e11]               [sigma11]
+    // depsilon = [e22]  ,   dsigma = [sigma22]
+    //            [e12]               [sigma12]
+    //            [e02]               [sigma02]
+    //            [e01]               [sigma01]
+    //
+    // mu = E / (2 * (1 + nu))
+    // bulk modulus = E / (2 * (1 - 2 * nu))
+    // lambda = (3 * bulk modulus - 2 * mu) / 3
+    // e_v = volumetric strain = e00 + e11 + e22
+    // lambda bar = lambda / (1 + e_v)
+    //
+    // dSigma = S * epsilon
+    //
+    // S_ijkl = lambda bar * delta_ij * delta_kl + 2 * mu * delta_ik * delta_jl
+    //
+    // Matrix form:
+    //
+    //     [2 mu + lambda bar     lambda bar         lambda bar                       ]
+    //     [   lambda bar      2 mu + lambda bar     lambda bar                       ]
+    //     [   lambda bar         lambda bar      2 mu + lambda bar                   ]
+    // S = [                                                           mu             ]
+    //     [                                                                 mu       ]
+    //     [                                                                       mu ]
+    //
+    // Above Voigt Notation is placed in a 3x3 matrix:
+    const CeedScalar strain_vol = gradu[0][0][i] + gradu[1][1][i] + gradu[2][2][i];
+    const CeedScalar lambda_bar = lambda / (1 + strain_vol);
+
+    const CeedScalar dsigma00 = (TwoMu + lambda_bar)*de[0][0] + lambda_bar*de[1][1] + lambda_bar*de[2][2],
+                     dsigma11 = lambda_bar*de[0][0] + (TwoMu + lambda_bar)*de[1][1] + lambda_bar*de[2][2],
+                     dsigma22 = lambda_bar*de[0][0] + lambda_bar*de[1][1] + (TwoMu + lambda_bar)*de[2][2],
+                     dsigma12 = 0.5*TwoMu*de[1][2],
+                     dsigma02 = 0.5*TwoMu*de[0][2],
+                     dsigma01 = 0.5*TwoMu*de[0][1];
```

Comment:
Is this Jacobian really correct (Newton converges quadratically)? I think we're getting confused with the factor of 2 that Voigt inserts in the shear components of `\epsilon` (but not `\sigma`). I think it should look like this (using Voigt's ordering convention, but not scaling).
```
                     dsigma12 = TwoMu*de[1][2],
                     dsigma02 = TwoMu*de[0][2],
                     dsigma01 = TwoMu*de[0][1];
```

Reply:
Note that you can do these checks automatically with DMSNESCheckFromOptions()

Reply:
@jeremylt This is the indicial notation of the Jacobian:
![hyperSS_dirv](https://user-images.githubusercontent.com/10523693/77375916-84aeda00-6d34-11ea-9bf6-ec02e6266c7f.png)

Reply:
Thanks, @knepley.

@jeremylt @ArashMehraban I somewhat hate the matrix format for displaying elasticity tensors, at least as a basis for actually computing. (It's good for clarity when explaining the formulation.) The reason being that it encourages code like the above instead of writing the derivative incrementally:
```
  lambda_dtrace = lambda_bar * (de[0][0] + de[1][1] + de[2][2]);
  dsigma00 = lambda_dtrace + TwoMu * de[0][0];
  dsigma11 = lambda_dtrace + TwoMu * de[1][1];
  dsigma22 = lambda_dtrace + TwoMu * de[2][2];
  dsigma12 = TwoMu * de[1][2];
  dsigma02 = TwoMu * de[0][2];
  dsigma01 = TwoMu * de[1][2];
```
which is much closer to the residual evaluation code and what you'd prefer the compiler to do anyway.

Reply:
Yep, @ArashMehraban, or in incremental notation
```
  dS = \bar\lambda \trace d\epsilon I + 2 \mu d\epsilon
```

Reply:
Now that we learned how code the Jacobian in HyperFS,  I totally agree with you that the matrix format sucks. I learned it from Engineering dept. and havebeen coding everything like that so far. I am switching to the way we coded HyperFS. Just out of curiosity, do you see anything wrong (from mathematical point of view) in the derivation of Jacobian in HyperSS?

Reply:
@jedbrown @jeremylt code it like this?
![hyperSS-linearization](https://user-images.githubusercontent.com/10523693/77384797-ec245400-6d4b-11ea-95b4-70fe5c64c975.png)

Reply:
This was it.

Reply:
Yeah, but you're mixing `\delta u` with `d u` and your equation (1) is mangled; just start with (2).

Note: I used `\operatorname{d} u` in the .rst, so it's typeset in upright and has suitable spacing on the left. I have often used `\delta u` in the context of Newton linearization (because the step isn't infinitesimal), but that sort of conflicts with the Kronecker delta already in the doc, so I switched to `d`.

Reply:
Thank you for catching that @jedbrown.